packages/ra-ui-materialui/src/input/CheckboxGroupInput.js

```diff
@@ -91,160 +83,116 @@ const styles = theme =>
-
-        return (
-            <FormControlLabel
-                htmlFor={`${id}_${get(choice, optionValue)}`}
-                key={get(choice, optionValue)}
-                checked={
-                    value
-                        ? value.find(v => v == get(choice, optionValue)) !== // eslint-disable-line eqeqeq
-                          undefined
-                        : false
-                }
-                onChange={this.handleCheck}
-                value={String(get(choice, optionValue))}
-                control={
-                    <Checkbox
-                        id={`${id}_${get(choice, optionValue)}`}
-                        color="primary"
-                        className={classes.checkbox}
-                        {...options}
-                    />
-                }
-                label={
-                    translateChoice
-                        ? translate(choiceName, { _: choiceName })
-                        : choiceName
-                }
-            />
-        );
-    };
+const CheckboxGroupInput = ({
+    choices,
+    helperText,
+    label,
+    onBlur,
+    onChange,
+    onFocus,
+    optionText,
+    optionValue,
+    options,
+    resource,
+    source,
+    translate,
+    translateChoice,
+    validate,
+    ...rest
+}) => {
+    const classes = useStyles();
 
-    render() {
-        const {
-            choices,
-            className,
-            classes = {},
-            isRequired,
-            label,
-            meta,
-            resource,
-            source,
-            input,
-            ...rest
-        } = this.props;
-        if (typeof meta === 'undefined') {
-            throw new Error(
-                "The CheckboxGroupInput component wasn't called within a react-final-form <Field>. Did you decorate it and forget to add the addField prop to your component? See https://marmelab.com/react-admin/Inputs.html#writing-your-own-input-component for details."
-            );
-        }
+    const {
+        id,
+        input: { onChange: finalFormOnChange, onBlur: finalFormOnBlur, value },
+        isRequired,
+        meta: { error, touched },
+    } = useInput({
+        onBlur,
+        onChange,
+        onFocus,
+        resource,
+        source,
+        validate,
+    });
 
-        const { touched, error, helperText = false } = meta;
+    const handleCheck = useCallback(
+        (event, isChecked) => {
+            let newValue;
+            try {
+                // try to convert string value to number, e.g. '123'
+                newValue = JSON.parse(event.target.value);
+            } catch (e) {
+                // impossible to convert value, e.g. 'abc'
+                newValue = event.target.value;
+            }
+            if (isChecked) {
+                finalFormOnChange([...(value || []), ...[newValue]]);
+            } else {
+                finalFormOnChange(value.filter(v => v != newValue)); // eslint-disable-line eqeqeq
```

Comment:
Idem, why use lose equality? I don't understand

Reply:
It was already the case in the previous version. I try to not alter how it worked. If you want to know why it's done like this, I suggest checking the file history ;)

Reply:
I think that's because values in form are strings, whereas our choices might be numbers, etc.